src/shared/specifier.c

```diff
@@ -120,7 +122,10 @@ int specifier_machine_id(char specifier, const void *data, const void *userdata,
         char *n;
         int r;
 
-        r = sd_id128_get_machine(&id);
+        if (empty_or_root((const char *) data))
+                r = sd_id128_get_machine(&id);
+        else
+                r = id128_read_prefix("/etc/machine-id", (const char *) data, ID128_PLAIN, &id);
```

Comment:
the cast is unnecessary, `const void*` implicitly casts to all other `const` pointer types

Reply:
Ok, C++ guy here. Seems like something a static analysis tool might get mad about